src/kixi/hecuba/weather/service.clj

```diff
@@ -1,24 +1,50 @@
 (ns kixi.hecuba.weather.service
   (:require [clojure.java.io :as io]
-            [clojure.tools.cli :refer [parse-opts]])
+            [clojure.tools.cli :refer [parse-opts]]
+            [clj-http.client :as client]
+            [clojure.data.json :as json]
+            [clojure.string :as str])
   (:gen-class))
 
 (def cli-options
   [["-c" "--config CONFIG" "Path to config edn file."
-    :default "config/settings.edn"]])
+    :default "config/settings.edn"]
+   ["-u" "--username USERNAME" "Your username."]
+   ["-p" "--password PASSWORD" "Your password."]])
 
 (defn load-config [filename]
   (with-open [r (io/reader filename)]
     (read (java.io.PushbackReader. r))))
 
-(defn find-entities [project-id api-endpoint entity-type]
-  (println project-id)
-  (println api-endpoint)
-  (println entity-type))
+(defn run-api-search [username password project-id api-endpoint entity-type max-entries-per-page]
```

Comment:
That's a good point, that function only started off with args to start off with then got bigger.